common/src/main/scala/org/apache/spark/sql/comet/execution/shuffle/ArrowReaderIterator.scala

```diff
@@ -42,10 +46,12 @@ class ArrowReaderIterator(channel: ReadableByteChannel, source: String)
     // memory leak.
     if (currentBatch != null) {
       currentBatch.close()
+      currentBatch = null
     }
 
     batch = nextBatch()
     if (batch.isEmpty) {
+      close()
```

Comment:
This is because I am paranoid and want to close the iterator early. I can revert changes to this file if you don't like it :)